test/run_as_user/run_as_user_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package run_as_user
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/test"
+)
+
+const (
+	// Let the agent run for 15 seconds. This will give agent enough time to change user
+	agentRuntime     = 15 * time.Second
+	configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+	pidFile          = "/opt/aws/amazon-cloudwatch-agent/var/amazon-cloudwatch-agent.pid"
+	root             = "root"
+	cwagent          = "cwagent"
+)
+
+type input struct {
+	user      string
+	dataInput string
+}
+
+func TestBundle(t *testing.T) {
+
+	parameters := []input{
+		{dataInput: "resources/default.json", user: root},
+		{dataInput: "resources/root.json", user: root},
+		{dataInput: "resources/cwagent.json", user: cwagent},
+	}
+
+	for _, parameter := range parameters {
+		t.Run(fmt.Sprintf("resource file location %s user %s", parameter.dataInput, parameter.user), func(t *testing.T) {
+			test.CopyFile(parameter.dataInput, configOutputPath)
+			test.StartAgent(configOutputPath, true)
+			time.Sleep(agentRuntime)
+			log.Printf("Agent has been running for : %s", agentRuntime.String())
+			// Must read the pid file while agent is running
+			pidOutput := test.RunCommand(test.CatCommand + pidFile)
```

Comment:
Personally, `runCommand` is generic for me enough to run a command. Therefore, I don't think we need a generic variable for it too. So a command would be enough `cat /opt/aws/amazon-cloudwatch-agent/var/amazon-cloudwatch-agent.pid` for a passing input here

Reply:
I like having cat as a const since it should be reusable.